lib/chore_runner/downloads/temporary_disk_storage_service.ex

```diff
@@ -0,0 +1,93 @@
+defmodule ChoreRunner.Downloads.TemporaryDiskStorageService do
+  alias ChoreRunner.Downloads.StorageService
+  @behaviour StorageService
+  @file_dir_name "chore_runner_temporary_disk_storage_service_directory"
+
+  @impl StorageService
+  def save_file(name, opts \\ []) do
+    with :ok <- ensure_dir(),
+         id = ChoreRunner.gen_id(),
+         file = %{
+           id: id,
+           name: name,
+           type: MIME.from_path(name),
+           path: build_write_filepath(id, name)
+         },
+         :ok <- write_file(file.path, opts[:path], opts[:body]) do
+      {:ok, Map.put(file, :created_at, File.stat!(file.path).mtime)}
+    end
+  end
+
+  @impl StorageService
+  def list_files do
+    case File.ls(dir()) do
+      {:ok, files} ->
+        {:ok, Enum.map(files, &build_file_from_saved_path/1)}
+
+      {:error, :enoent} ->
+        {:ok, []}
+
+      error ->
+        error
+    end
+  end
+
+  @impl StorageService
+  def delete_file(%{path: path} = _file), do: File.rm(path)
```

Comment:
From Dialyzer
```
The inferred type for the 1st argument of delete_file/1 (
          #{'path' :=
                binary() |
                maybe_improper_list(binary() |
                                    maybe_improper_list(any(),
                                                        binary() | []) |
                                    char(),
                                    binary() | []),
            _ => _}) is not a supertype of 
          #{'chore_id' := binary(),
            'created_at' :=
                #{'__struct__' := 'Elixir.DateTime',
                  'calendar' := atom(),
                  'day' := pos_integer(),
                  'hour' := non_neg_integer(),
                  'microsecond' :=
                      {non_neg_integer(), non_neg_integer()},
                  'minute' := non_neg_integer(),
                  'month' := pos_integer(),
                  'second' := non_neg_integer(),
                  'std_offset' := integer(),
                  'time_zone' := binary(),
                  'utc_offset' := integer(),
                  'year' := integer(),
                  'zone_abbr' := binary()},
            'id' := binary(),
            'name' := binary(),
            'type' := binary()}, which is expected type for this argument in the callback of the 'Elixir.ChoreRunner.Downloads.StorageService' behaviour
```

Reply:
While this makes sense, if there's a different way to typespec this then I am all ears. the file typespec is a map with the required keys id, name, type, and created_at. However implementers of the `StorageService` behaviour can add whatever keys they need to it, it's not a struct, and `TemporaryDiskStorageService` uses the `path` key, among others. So while it looks like it betrays the spec, it's valid.

Reply:
Can change the typespec to 
```elixir
    @type file :: %{
          required(:id) => String.t(),
          required(:name) => String.t(),
          required(:type) => String.t(),
          required(:created_at) => DateTime.t(),
          optional(:chore_id) => String.t(),
          optional(atom()) => any()
        }
```